### PR TITLE
Bugfix/#64: Empty regions verification optimization

### PR DESCRIPTION
**Pull request by GyD**
This pull request relates to [BOSA_0100-358](https://jira.hosted-tools.com/browse/BOSA_0100-358) (JIRA issue) and contains Yves's (@GyD) attempt to improve the performance of Kiso.

- The `has_{region}` functionality is kept so theme using it still can use it. Functions used to check if region is empty are kept but adapted so theme using it still can use it.
- A per region variable `page_{region}` is introduced for future themes, that variable store the rendered output of the `page.region` so it can be called in theme without re-redering the region again,
- A function `_kiso_render_region()` is introduced to avoid calling `\Drupal::service('renderer')->render($region);` each time *but* also not mark the region as rendered since the region is not passed by reference. This way if you need to alter region content after `kiso_preprocess_html()` and `kiso_preprocess_page()` it still behave as Drupal core.

### CODE REVIEW

File: includes/preprocess.utils.inc [hunk 1]
```diff
@@ -1,14 +1,77 @@
...
+  }
+}
+
 /**
  * Strips html tags, except allowed, returning a trimmed clean markup.
  */
 function _kiso_strip_tags(string $markup, string $allowed_tags = '') {
-  $allowed_tags .= '<drupal-render-placeholder><img>';
-  return trim(strip_tags($markup, $allowed_tags));
+  $text = strip_tags($markup, [
```
Comment (rutiolma) [inline includes/preprocess.utils.inc:61]: It's ignoring the allowed_tags param now but we should still support it

File: kiso.info.yml [hunk 2]
```diff
@@ -58,7 +58,3 @@ libraries-override:
     css:
       theme:
         extlink.css: 'css/components/extlink/extlink-window.css'
-
```
Comment (rutiolma) [inline kiso.info.yml:61]: I would keep this, so we start using it on new projects.
This module will be added on the next release of Openfed

**Review by rutiolma — APPROVED** [review]
Looks good to me

File: includes/preprocess.html.inc [hunk 3]
```diff
@@ -9,6 +9,32 @@
  * Implements hook_preprocess_HOOK() for html.html.twig.
  */
 function kiso_preprocess_html(&$variables) {
+
+  $variables['page_tools'] = _kiso_render_region($variables['page']['tools']);
```
Comment (smillart) [inline includes/preprocess.html.inc:13]: @GyD What's the purpose of this `$variables['page_tools']` variable defined here from the `kiso_preprocess_html()` function? I cannot see any `page_tools` variable called inside the `html.html.twig` template...

**Review by smillart — CHANGES_REQUESTED** [review]
Just a comment added regarding the `kiso_preprocess_html()` function that contains a potentially unused `$variables['page_tools']` variable. @GyD  Could you please take a look? Thanks!

**Review by smillart — APPROVED** [review]

**Pull request merged**